pennylane/interfaces/execution.py

```diff
@@ -363,15 +367,25 @@ def cost_fn(params, x):
             device_batch_transform=device_batch_transform,
         )
 
+    new_device_interface = isinstance(device, qml.devices.experimental.Device)
+
     if interface == "auto":
         params = []
         for tape in tapes:
             params.extend(tape.get_parameters(trainable_only=False))
         interface = qml.math.get_interface(*params)
+    config = qml.devices.experimental.ExecutionConfig(interface=interface)
 
     gradient_kwargs = gradient_kwargs or {}
 
-    if device_batch_transform:
+    if new_device_interface:

```

Comment:
Given the number of conditions I would create a private function and call it inside the main. It will make easier to work with and much more readable.